### DISCUSSION TITLE
Best practices for managing Mongoose models and migrations?

### DISCUSSION THREAD

**Discussion by AxelBriche** [Automattic/mongoose#13064] [Q&A]
Hey everyone,

I've been working with Mongoose, and I've run into some issues with managing my models and migrations. Specifically, I'm running into issues with duplicate errors and I'm not quite sure how to handle migrations in the most efficient way.

I've tried doing some research on best practices, but I haven't been able to find a lot of helpful information. So, I wanted to reach out to the community and see if anyone has any tips or resources they could recommend.

How do you manage your Mongoose models to avoid duplicate errors? And what's the best way to handle migrations when you need to make changes to your schema? Any advice or insights would be greatly appreciated!

Thanks in advance.

**Comment by vkarpov15** [comment]
Can you provide an example of a migration you're running that you're running which causes a duplicate error?

We unfortunately don't have much in the way of documentation on this front. We've been occasionally working on a [migration package](https://www.npmjs.com/package/@mongoosejs/migrations), but it is currently only in internal alpha testing. [Here's a talk I gave at MongoDB World about the general idea of the migration framework](https://docs.google.com/presentation/d/1jkvUAErLFgGpImjWPMT-L1RyPfLPoCMeyVqAZyckZXE/edit?usp=sharing), does that sound interesting?
